[Do Not Merge] - [WIP] OpenJDK 11 Upgrades

rjhancock opened the pull request:
Working through the warnings, errors, and notices in VS Code to prep the codebase for a transition to OpenJDK 11.

Each commit is limited to each file, if possible. (apologies for the commit messages as they should have been more verbose from the beginning).

GalaxyCr8r committed:
Update README.md

We should REALLY specify which version of Java we are targeting, or at least using for development. Luckily the current version of OpenJDK worked for me, but we should still note it somewhere.

WEKarnesky committed:
Merge pull request #218 from Galaxy613/patch-1

Update README.md

rjhancock committed:
Added Ignores for paths/files added by VSCode

rjhancock committed:
Type restrict Vectors
Long() -> Long.valueOf()
Type restrict Hashtable

rjhancock committed:
Added serialVersionUID
Added types to Tempaltes

rjhancock committed:
Deprication removals
Serial IDs
Template Declarations with Types

rjhancock committed:
Java Warnings related to arrasy and unchecked types

rjhancock committed:
Strict Type declarations.
Commented out code that is not needed.
removed import not being used.
Updated Advantages to conform to type.

rjhancock committed:
Commented out code not used
Static types for tempaltes

rjhancock committed:
Strict type control
Commenting out code not being used.

rjhancock committed:
Type enforcement for Templates
Corrected type for abUnitData

rjhancock committed:
Commented out code that doesn't appear to be used.
Type enforcement for methods

rjhancock committed:
Static template adjustments

rjhancock committed:
Static type declaration for templates

rjhancock committed:
Static Assignment of Types for Templates

rjhancock committed:
Serial Version UID added to class

rjhancock committed:
serial Version UID Added

rjhancock committed:
Serial Version UID and Removed warning supression

rjhancock committed:
Serial Version UID Added

rjhancock committed:
Commented out code not being used
Added Serial Version ID
Removed warning supression

rjhancock committed:
Updated gitignore for VS Code settings

rjhancock committed:
Commented out unused variable

rjhancock committed:
Added Serial Version UID

rjhancock committed:
Removed warning supression and added Serial Version UID

rjhancock committed:
Added serial version UID

rjhancock committed:
Added Serial Version UID

rjhancock committed:
SAW/GUI/dlgOpen - Type enforcement of combo boxes.

rjhancock committed:
SAW/GUI - dlgPlaceableInfo
- Removed warning supression
- Added Serial Version UID

rjhancock committed:
SAW/GUI - dlgSummaryInfo
- Added Serial Version UID

rjhancock committed:
SAW/GUI - dlgTechBaseChooser
- Added Serial Version UID
- Removed warning supression

rjhancock committed:
SAW/GUI - dlgTextExport
- Added Serial Version UID
- Removed warning supression

rjhancock committed:
SAW/GUI - dlgVariableSize
- Added Serial Version UID
- Removed Warning Supression

rjhancock committed:
SAW/GUI - dlgWeaponInfo
- Aded Serial Version UID

rjhancock committed:
SAW/GUI - EquipmentListRenderer
- Commented out values not being used
- added Serial Version UID
- Added comment regarding parameter and type

rjhancock committed:
SAW/GUI - EquipmentSelectedRenderer
- Added Serial Version UID
- Added comment regarding list being passed in

rjhancock committed:
SAW/GUI - ifVeeForm
- Removed imports not being used

rjhancock committed:
SAW/GUI - EquimentListRenderer/EquipmentSelectedRenderer
- Adjusted type of JList to be "Any"

rjhancock committed:
SSW/Components - CCGMech
- Removed imports that weren't used.
- Commented out variables not used
- Commented out loops that weren't used.
- Additional formmating done via VSCode

rjhancock committed:
ssw/filehandlers - HMPReader
- Added comment regarding 'unused' variable

rjhancock committed:
ssw/filehandlers - MTFReader
- Commented out variables not being used.

rjhancock committed:
ssw/gui - DamageChart
- Added Serial Version UID
- Typed Vector Templates.

rjhancock committed:
ssw/gui - dlgAboutBox
- Added Serial Version UID
- VS Code Code Formating.

rjhancock committed:
ssw/gui - dlgAmmoLotSize
- Added Serial Version UID
- VS Code Formatting

rjhancock committed:
ssw/gui - dlgArmorTonnage
- Added Serial Version UID
- VS Code Formatting

Algebro7 committed:
save reference to mech quirk list

Algebro7 committed:
move quirks assignment to a more logical location

WEKarnesky committed:
Merge pull request #223 from Solaris-Skunk-Werks/bugfix/save-quirks

Fix issue saving quirks on new mechs

Algebro7 commented on ssw/src/main/java/ssw/filehandlers/MTFReader.java:
```diff
@@ -29,21 +29,21 @@
+// import java.util.Hashtable;
+// import visitors.VMechFullRecalc;
 
 public class MTFReader {
```
I think this whole class can actually be removed since I don't think it's being used anywhere

Algebro7 commented on ssw/src/main/java/ssw/gui/DamageChart.java:
```diff
@@ -37,10 +37,15 @@
     private int GridY = 1;
-    private Vector charts = new Vector();
-    private Vector colors = new Vector();
+    private Vector<int[]> charts = new Vector<int[]>();
```
I think you can do `private Vector<int[]> charts = new Vector<>();` instead, the type can be omitted in the constructor

Algebro7 commented on ssw/src/main/java/ssw/gui/DamageChart.java:
```diff
@@ -37,10 +37,15 @@
-    private Vector charts = new Vector();
-    private Vector colors = new Vector();
+    private Vector<int[]> charts = new Vector<int[]>();
+    private Vector<Color> colors = new Vector<Color>();
```
Same here

Algebro7 commented on BFB/src/main/java/BFB/GUI/ForcePanel.java:
```diff
@@ -93,7 +93,7 @@ private void initComponents() {
         jLabel1 = new javax.swing.JLabel();
         txtForceName = new javax.swing.JTextField();
-        cmbForceType = new javax.swing.JComboBox();
+        cmbForceType = new javax.swing.JComboBox<String>();
```
So I found a way to fix these in the form designer, which will allow the updates to persist when editing any dialogs in Netbeans. For example, open BFB/GUI/ForcePanel.java in Netbeans, right click the cmbJForceType component, go to Code, and then type `<String>` in the "Type Parameters" box and then save the form:

![typeproperties](https://user-images.githubusercontent.com/5447759/102571559-ae3d3c00-40e1-11eb-9a2c-d162ac30d56b.png)

Can you do these for any changes that take place in autogenerated code sections, instead of modifying the Java directly?

Algebro7 replied:
You shouldn't need to revert or undo the previous changes, because once you open and save the forms in Netbeans the changes in the autogenerated code sections will be erased anyway

rjhancock replied:
Not using NetBeans. Using VS Code. Although that is good to know for future forms y'all mess with. :)

Algebro7 replied:
Unfortunately Netbeans is required for any changes to GUI code :( I wish there was a way to make it totally IDE-agnostic but it would be too hard for us not to have a form designer.

I only use Netbeans for that purpose, any IDE will work for logic changes that don't happen inside of autogenerated GUI code

Algebro7 commented on .gitignore:
```diff
@@ -91,3 +91,9 @@ gradle-app.setting
 # End of https://www.gitignore.io/api/gradle,intellij
 Logs/
+
+# VSCode Adds
```
Should we add .vscode also? Or is there helpful data in there that should be checked into source control?

rjhancock replied:
I didn't want to pollute with the VS Code stuff. But I can see if there is anything of use in them.

Algebro7 replied:
Probably best to just exclude `.vscode` here to keep IDE-specific stuff out as much as possible

Algebro7 commented on BFB/src/main/java/BFB/GUI/ForcePanel.java:
```diff
@@ -39,6 +39,7 @@
 import javax.swing.event.TableModelListener;
 
 public class ForcePanel extends javax.swing.JPanel {
+    private static final long serialVersionUID = 9012868947346629504L;
```
Just out of curiosity, how were these generated?

rjhancock replied:
VS Code has a 'basic' or 'random' option and just let it pick it at random.

rjhancock committed:
Upstream Master Merge

rjhancock committed:
Commented out variables and methods not used in Class.

rjhancock committed:
sswlib/src/main/java/visitors - VHeatSinkSetProtoDouble
- Commented out varible not being used
- VS Code Formatting.

rjhancock committed:
sswlib/src/main/java/visitors - VGyroSetNone.java
- Commented out import not being used.
- VS Code Formatting.

rjhancock committed:
sswlib/src/main/java/visitors - VFCSArtemisVLoader
    - Type Enforcement of ArrayList;s
    - Update Interfaces to enforce Type Enforcement
    - Fix compile errors related to changes
    - VS Code Formatting.

rjhancock committed:
sswlib/src/main/java/visitors - VFCSArtemisIVLoader
- Strick Type Enforcement
- VC Code Formatting

rjhancock committed:
sswlib/src/main/java/visitors - VFCSApolloLoader
- Type Enforcement
- VS Code Formatting

rjhancock committed:
sswlib/src/main/java/visitors - VArmorSetVStealth
- Commented out variables not being used.

rjhancock committed:
sswlib/src/main/java/visitors - VArmorSetPatchworkLocation
- Type Enforcement
- VS Code Formatting

rjhancock committed:
sswlib/src/main/java/visitors - VArmorSetIR
- Commented out variable not being used

rjhancock committed:
sswlib/src/main/java/visitors
- Commented out variable not being used.

rjhancock committed:
sswlib/src/main/java/visitors - VArmorSetBR
- Commented out variables not being used

rjhancock committed:
sswlib/src/main/java/visitors - VArmorSetAB
- Commented out variable not in use
- VS Code Formatting

rjhancock committed:
sswlib/src/main/java/utilities - CVCostBVBreakdown
- Strict Type Enforcement
- Commented out variables not being used
- VS Code Formatting

rjhancock committed:
sswlib/src/main/java/utilities - CostBVBreakdown
- Strict Type Enforcement

rjhancock committed:
sswlib/src/main/java/states - stEngineNone
- Commented out methods not being used

rjhancock committed:
sswlib/src/main/java/states - stChassisCVIS
- Commented out method not being used

rjhancock committed:
sswlib/src/main/java/Print - TWVTOLPoints
- Commented out variable not being used

rjhancock committed:
sswlib/src/main/java/Print - TWNavalPoints
- Commented out variables not being used.

rjhancock committed:
sswlib/src/main/java/Print - TWAdvGroundPoints
- Commented out variables not being used.

rjhancock committed:
sswlib/src/main/java/print - ScenarioPrinter
- Commented out variables not being used.

rjhancock committed:
sswlib/src/main/java/Print - QuickStrieCardPrinter
- Commented out variables/import not being used

rjhancock committed:
sswlib/src/main/java/print - QSVerticalCardPrinter
- Commented out variables not being used
- Added brackets to If statements to clear up what code gets ran

rjhancock committed:
sswlib/src/main/java/print - QSHorizontalCardPrinter
- Commented out variables not being used.

Algebro7 committed:
reset quirks when clicking new mech button

Algebro7 committed:
fix indentation

Algebro7 committed:
set curlotsize when deserializing Ammunition

rjhancock committed:
ssw/filehandlers/HMPReader.java
- Adjusted unused variable to not be needed.
- VS Code Formatting

rjhancock committed:
ssw/gui/dlgBatchHMP.java
- Added Serial Version
- Commented out variable not being used
- VS Code Formatting

rjhancock committed:
ssw/gui - dlgBracketCharts.java
- Added serialization IDs
- Strict typing of template variables
- VS Code Styling

rjhancock committed:
Mass Update - serialVersionUID Added

rjhancock committed:
ssw/gui/dlgCCGMech.java
- Commented out unused variables
- Adjusted variable types to prevent unchecked casts.

ssw/components/CCGMech.java
- Adjusted method declaration and type conversion to better conform with intent.

rjhancock committed:
print/QSCardSheetPrinter.java
- Removed variable not in use

rjhancock committed:
Mass Update - Removed SuprressWarnings

rjhancock committed:
Single Warning Issues Fixed
- Unused Variables
- Variable Template Restrictions
- Remove unused Imports
- Removed null referenceing
- Writer Closing
- Removed duplicate implements
- Removed assignment to variables not used

rjhancock committed:
2 Warning Files
- Unused Variables
- Type Saftey
- Removed unused imports

rjhancock committed:
3 Warning Fixes
- Type Enforcement
- Commented out code not being used
- Commented out unused variables
- Removed unused Imports
- Ensure Buffer Closes

rjhancock committed:
Mass Update - Import not used

rjhancock committed:
Print - PrintConsts.java

rjhancock committed:
Ensure files are closed

rjhancock committed:
ssw/gui - dlgPrintBatchMechs.java
- Removed constructor not being used in program
- Type enforcement for tempaltes

rjhancock committed:
ssw/print - Printer.java
- Removed unused Variables
- Enforced templated types
- VS Code Formating

rjhancock committed:
ssw/printpreview - Preview.java
- Removed unused variables
- remvoed noop code.
- adjusted assignment to a variable that isn't used

rjhancock committed:
battleforce - BattleForce.java
- Commented out variables not in use
- Commented out private methods not in use.

rjhancock committed:
battleforce - BattleForceData.java
- Commented variables not in use
- VS Code Formatting

rjhancock committed:
components - CVAmmunitionHandler
- Removed unchecked Conversion warning

rjhancock committed:
components - CVHeatSinkFactory
- Type enforcement
- Removed unused variables
- VS Code Formatting

rjhancock committed:
components - CVJumpJetFactory.java
- Forced Type Linking
- VS Code Formatting.

rjhancock committed:
components - EquipmentCollection
- Type enforcement
- Removed unneeded type conversion
- VS Code Styling

rjhancock committed:
components - HeatSinkFactory
- Enforced Tempalte Types
- Removed unneeded type conversions
- VS Code Formatting

rjhancock committed:
components - ifCVLoadout
- Type enforcement for templates
- VS Code Formatting

rjhancock committed:
components - ifMechLoadout
- Template Type Enforcement

rjhancock committed:
sswlib/src/main/java/Print - PrintVehicle.java
- Removed unused variables and private methods

rjhancock committed:
sswlib/src/main/java/Print - PrintMech
- Template Type Enforcement
- Removed variables and methods not used
- VS Code Formatting

rjhancock committed:
sswlib/src/main/java/Print/preview/PreviewDialog
- Removed methods not used locally
- VS Code Formatting

WEKarnesky committed:
Merge pull request #228 from Solaris-Skunk-Werks/bugfix/ammo-lot-size

Set curlotsize when deserializing Ammunition

WEKarnesky committed:
Merge pull request #227 from Solaris-Skunk-Werks/bugfix/save-quirks

Reset quirks when clicking new mech button

rjhancock committed:
Merge branch 'develop' into jdk11

gblouin committed:
Handle Issues with Paths

That are stored in the file but do not exist on the current users system.

WEKarnesky committed:
Merge pull request #231 from Solaris-Skunk-Werks/imagefix

Handles Issues with SSW Image Paths

gblouin committed:
Set BV of Capacitor for CL ER PPC

Since we have a specific reference for this total value.

gblouin committed:
BV Updates

Calculate the offset for known values from rulebooks but keep the original multiplier code for those that don't have set values yet.

gblouin committed:
Use EncodeFluff for MUL

This section was missed when the EncodeFluff method was added.  This will now convert " into &quot; along with other conversions necessary for XML.

gblouin committed:
Convert Double Quotes to Single

So that they will match up with what Megamek is doing.

WEKarnesky committed:
Merge pull request #232 from Solaris-Skunk-Werks/CLERPPC_CapacitorBV

PPC Capacitor BV Update

WEKarnesky committed:
Merge pull request #233 from Solaris-Skunk-Werks/MUL_Export

Convert double quotes to single quotes during MUL export

gblouin committed:
Instantiate each view once

and then just choose the one to use and fill it with units.   Hoping that this clears up Maelwys's issues with the rendering being slow.

gblouin committed:
Encode to UTF-8

as I was getting build errors for smart quotes in the advantages file.

gblouin committed:
Update Usages

From other areas too

gblouin committed:
Missed file

Updated to new constructor

gblouin committed:
VSCode Space Changes

My editor seems to remove extra spaces from the ends of lines

gblouin committed:
Make the Prefs use Vee

Instead of Mech since this is for SAW

gblouin committed:
GUI Corrections

Use SAW Prefs and make the options match.
Clear the "Selected Variant" dropdown.
Manage the Lock Chassis button correctly.

gblouin committed:
Handle Long Unit/Model Names

For the Roster sheet, this will break up a long name into two lines.
Also pushed the tonnage away from the unit type a bit more so that "Combat Vehicle" wouldn't overlap.

gblouin committed:
Handle Mechwarrior Names Too

And now can handle multiple lines for each and then adds just a bit more space under the line so it doesn't feel quite so squished.

gblouin committed:
Open Vehicle work

So that it sets the buttons and comboboxes correctly.

gblouin committed:
Stop Enabling

When it shouldn't be enabled.

gblouin committed:
Hitting New wasn't clearing Variant

I had slightly different code, now matching what the other code is doing.

WEKarnesky committed:
Merge pull request #237 from Solaris-Skunk-Werks/bfb_roster

Updated Roster Printing Code

WEKarnesky committed:
Merge pull request #234 from Solaris-Skunk-Werks/selectunits

Select Unit screen update

WEKarnesky committed:
Merge pull request #236 from Solaris-Skunk-Werks/sawgui

SAW GUI Fixes

gblouin committed:
Add button to convert to MTF

Will load and process all files in the directory (and any subdirectory)

gblouin committed:
Make the file count for Progress consistent

So that it doesn't jump around everywhere.

gblouin committed:
Changing Icon to Crown

gblouin committed:
Check Message for Empty

and if it is...do not show the message pop-up.

gblouin committed:
Take this Gradle!

Convert from isBlank to a length check after trim.

gblouin committed:
Null check

before trying to access the height property.

WEKarnesky committed:
Merge pull request #238 from Solaris-Skunk-Werks/convertdirectory_tomtf

Add button to convert to MTF

Commit:
drafting release notes

WEKarnesky committed:
Merge pull request #241 from Solaris-Skunk-Werks/release-changes

drafting release notes

WEKarnesky committed:
Merge branch 'release/v0.7.6'

WEKarnesky committed:
Merge tag 'v0.7.6' into develop

WEKarnesky committed:
forgot to commit Version number before release

WEKarnesky committed:
Merge branch 'develop'

zophar committed:
fix file loading issue

WEKarnesky committed:
Merge pull request #244 from zophar/fix_linux_protocol_error

Fix protocol error on Linux when opening files

WEKarnesky committed:
Changed Clan SRM-2 Improved Oneshot weight to 0.25tons

WEKarnesky committed:
Merge branch 'hotfix/clanSRM2Weight'

rjhancock committed:
Merge remote-tracking branch 'upstream'

rjhancock committed:
Latest updates.
